test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnector.cs

```diff
@@ -18,60 +18,67 @@ namespace EdgeHubRestartTester
         Guid batchId;
-        DateTime runExpirationTime;
-        CancellationToken cancellationToken;
-        DateTime edgeHubRestartedTime;
-        uint restartSequenceNumber;
         ILogger logger;
 
+        ModuleClient DmModuleClient
+        {
+            get
+            {
+                if (this.dmModuleClient == null)
+                {
+                    this.dmModuleClient = ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.TransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        this.logger).Result;
+                }
+
+                return this.dmModuleClient;
+            }
+        }
+
         public DirectMethodEdgeHubConnector(
-            ModuleClient dmModuleClient,
             Guid batchId,
-            DateTime runExpirationTime,
-            DateTime edgeHubRestartedTime,
-            uint restartSequenceNumber,
-            ILogger logger,
-            CancellationToken cancellationToken)
+            ILogger logger)
         {
-            this.dmModuleClient = Preconditions.CheckNotNull(dmModuleClient, nameof(dmModuleClient));
             this.batchId = batchId;
-            this.runExpirationTime = runExpirationTime;
-            this.cancellationToken = Preconditions.CheckNotNull(cancellationToken, nameof(cancellationToken));
-            this.edgeHubRestartedTime = edgeHubRestartedTime;
-            this.restartSequenceNumber = restartSequenceNumber;
             this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
         }
 
-        public async Task StartAsync()
+        public void Dispose() => this.dmModuleClient?.Dispose();
+
+        public async Task StartAsync(
+            DateTime runExpirationTime,
+            DateTime edgeHubRestartedTime,
+            CancellationToken cancellationToken)
         {
             (DateTime dmCompletedTime, HttpStatusCode dmStatusCode) = await this.SendDirectMethodAsync(
                 Settings.Current.DeviceId,
                 Settings.Current.DirectMethodTargetModuleId,
-                this.dmModuleClient,
+                this.DmModuleClient,
                 Settings.Current.DirectMethodName,
-                this.runExpirationTime,
-                this.cancellationToken,
+                runExpirationTime,
+                cancellationToken,
                 this.logger).ConfigureAwait(false);
 
             TestResultBase dmTestResult = new EdgeHubRestartDirectMethodResult(
                 Settings.Current.ModuleId + "." + TestOperationResultType.EdgeHubRestartDirectMethod.ToString(),
                 DateTime.UtcNow,
                 Settings.Current.TrackingId,
                 this.batchId,
-                Interlocked.Read(ref this.directMethodCount).ToString(),
-                this.edgeHubRestartedTime,
+                this.directMethodCount.ToString(),
```

Comment:
what is the expected value of direct method count to report to TRC for each run?

Reply:
Increasing number ( not necessary incremental ).